testing/README.md

```diff
@@ -390,9 +393,55 @@ Visual regression testing allows you to be confident that changes made to the co
 [↑ scrollTo('#table-of-contents')](#table-of-contents)
 
 
-
 ## React
 
 If your project is using React, you can find React-specific testing recommendations in the [React styleguide](https://github.com/Shopify/javascript/tree/master/react#testing).
 
 [↑ scrollTo('#table-of-contents')](#table-of-contents)
+
+## Coverage
+
+Generating a test coverage report can help you identify areas of the code that are lacking tests. Coverage reports can be integrated with CodeCov and Circle CI to ensure sufficient coverage with every checkin. See the [CodeCov Integration Guide](https://vault.shopify.com/Codecov-integration) for more information.
+
+Here are some recommended ways to generate coverage:
+
+- **Karma**: You can use the [Karma Istanbul Reporter plugin](https://github.com/mattlewis92/karma-coverage-istanbul-reporter) to automatically generate coverage when tests are running.
+- **Jest**: You can use the built in coverage reporter by adding the `--coverage` argument when running tests. See (documentation)[https://facebook.github.io/jest/docs/en/cli.html#coverage].
+- **Other**: We recommend using [nyc](https://github.com/istanbuljs/nyc) as a standalone command line tool to generate coverage.
+
+It is best practice to generate coverage for all relevant JS files as opposed to only tested files. This will files are not inadvertently missed. There are a couple of ways to generate coverage for all files:
+- **Karma**: Add the [`includeAllSources`](https://github.com/karma-runner/karma-coverage/blob/master/docs/configuration.md#includeallsources) option to your Karma `coverageReporter` config:
+```js
+files: [
+  'app/assets/javascripts/**/*.js', // Regexp matching relevant files
+],
+coverageReporter: { 
+  includeAllSources: true
+}
+```
+- **Jest**: If you are using [Sewing-kit](https://github.com/Shopify/sewing-kit), coverage for all files should be enabled automatically. Otherwise, you can configure the [`collectCoverageFrom`](http://facebook.github.io/jest/docs/en/configuration.html#collectcoveragefrom-array) option:
```

Comment:
What is otherwise here? When is coverage not collected?

Reply:
I meant to say, if you're not using Sewing-kit, you might need to set the `collectCoverageFrom` option yourself.